docroot/modules/humsci/hs_capx/src/Form/CapxCredsForm.php

```diff
@@ -0,0 +1,130 @@
+<?php
+
+namespace Drupal\hs_capx\Form;
+
+use Drupal\Core\Cache\Cache;
+use Drupal\Core\Config\ConfigFactoryInterface;
+use Drupal\Core\Form\ConfigFormBase;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Url;
+use Drupal\hs_capx\Capx;
+use Drupal\key\Entity\Key;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Form class to set the CAPx user credentials.
+ *
+ * @package Drupal\hs_capx\Form
+ */
+class CapxCredsForm extends ConfigFormBase {
+
+  /**
+   * CAPx Service.
+   *
+   * @var \Drupal\hs_capx\Capx
+   */
+  protected $capx;
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container) {
+    return new static(
+      $container->get('config.factory'),
+      $container->get('capx')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function __construct(ConfigFactoryInterface $config_factory, Capx $capx) {
+    parent::__construct($config_factory);
+    $this->capx = $capx;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getFormId() {
+    return 'capx_creds_form';
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function getEditableConfigNames() {
+    return ['hs_capx.settings'];
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildForm(array $form, FormStateInterface $form_state) {
+    $form = parent::buildForm($form, $form_state);
+    $config = $this->config('hs_capx.settings');
+    $form['username'] = [
+      '#type' => 'textfield',
+      '#title' => $this->t('Username'),
+      '#required' => TRUE,
+      '#default_value' => $config->get('username'),
+    ];
+
+    $keys = Key::loadMultiple();
+    foreach ($keys as &$key) {
```

Comment:
Also the $key variable doesn't appear to be used.

Reply:
`$keys` is used as the options in the form, so what this is doing is changing the `$key` object into a string since the foreach is looping with reference.

Reply:
Ah, I missed that detail. Please add a code comment.

Reply:
Comment added